proof/refine/ARM/Detype_R.thy

```diff
@@ -100,6 +100,9 @@ defs deletionIsSafe_def:
        (\<forall>ko. ksPSpace s p = Some (KOArch ko) \<and> p \<in> {ptr .. ptr + 2 ^ bits - 1}
         \<longrightarrow> 6 \<le> bits)"
 
+defs deletionIsSafe2_def:
+  "deletionIsSafe2 \<equiv> \<lambda>ptr bits s. \<forall>p. ko_wp_at' live' p s \<longrightarrow> p \<notin> {ptr .. ptr + 2 ^ bits - 1}"
```

Comment:
I know this is coming in pretty late in the game, but we have this abbreviation `mask_range ptr bits` (in Word_Lib) for `{ptr .. ptr + mask bits}` which is equivalent to `{ptr .. ptr + 2 ^ bits - 1}` and has a bunch of nice automation setup. Would it be extremely annoying to update and use that one instead of the expanded form?

Reply:
Actually, I see that RISCV64 does use that one. Any specific reason why it's not used here?

Reply:
I can never keep track of which architecture uses which approach for ranges like this. I just followed along with what is used in `deletionIsSafe`. Are we supposed to have `mask_range` used everywhere on all arches?

Reply:
For any new uses, yes. The older architectures don't use it yet in their definitions, because it didn't exist yet when those proofs were written, but any new material should use `mask_range` when applicable.

Reply:
Alright. I can update these